packages/frontend/src/components/SimpleChart/index.tsx

```diff
@@ -180,6 +180,9 @@ const SimpleChart: FC<SimpleChartProps> = memo((props) => {
                                     // on `useEchartsCartesianConfig` to generate the tooltip
                                     if (eChartsOptions.tooltip.formatter) {
                                         const dim =
+                                            param.encode?.x &&
+                                            Array.isArray(param.encode?.x) &&
+                                            param.encode.x.length > 0 &&
                                             param.encode?.x[0] !== undefined
```

Comment:
This is good, 
Just wondering, this is the same as doing `encode.x?.[0] !== undefined` , right ? Not sure if there is any difference.

Reply:
Thanks for the suggestion @rephus !
I've checked and seems like `encode.x` will be an array if not undefined. 
Pushed that change here: https://github.com/lightdash/lightdash/pull/10076/commits/9b535ba3b6bc64225b366eaccfec5ccaa4d75968